test/libyul/Common.cpp

```diff
@@ -75,6 +75,27 @@ pair<shared_ptr<Block>, shared_ptr<yul::AsmAnalysisInfo>> yul::test::parse(strin
 	return make_pair(stack.parserResult()->code, stack.parserResult()->analysisInfo);
 }
 
+pair<shared_ptr<Block>, shared_ptr<yul::AsmAnalysisInfo>> yul::test::parse(
+	string const& _source,
+	Dialect const& _dialect,
+	ErrorList& _errors
+)
+{
+	ErrorReporter errorReporter(_errors);
+	shared_ptr<Scanner> scanner = make_shared<Scanner>(CharStream(_source, ""));
+	shared_ptr<Object> parserResult = yul::ObjectParser(errorReporter, _dialect).parse(scanner, false);
+	if (!parserResult)
+		return {};
+	if (!parserResult->code || !errorReporter.errors().empty())
+		return {};
+	shared_ptr<AsmAnalysisInfo> analysisInfo = make_shared<AsmAnalysisInfo>();
+	AsmAnalyzer analyzer(*analysisInfo, errorReporter, _dialect, {}, parserResult->dataNames());
+	// TODO this should be done recursively.
```

Comment:
TODO in this PR or next?

Reply:
Some other time. We currently do not have test cases that would require it.

Reply:
Would it make sense to add a test case now, even if it doesn't compile, just to be aware in the future?

Reply:
Or create a new issue maybe

Reply:
This is for nested objects. It will break, because there is no analysis. Added issue: https://github.com/ethereum/solidity/issues/8143